units/stats_task_end.go

```diff
@@ -130,8 +131,8 @@ func (j *collectTaskEndDataJob) Run(ctx context.Context) {
 		"version":              j.task.Version,
 	}
 
-	if j.task.DisplayTask != nil {
-		msg["display_task_id"] = j.task.DisplayTask.Id
+	if utility.FromStringPtr(j.task.DisplayTaskId) != "" {
```

Comment:
This won't be populated for old tasks but I'm assuming that's not expected by the user.

Reply:
Yes this is only required going forward